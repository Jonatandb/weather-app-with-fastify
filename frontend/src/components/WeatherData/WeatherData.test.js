const { render, screen } = require('@testing-library/react')
const { default: WeatherData } = require('./WeatherData')

describe('WeatherData', () => {
  it('should render correctly', () => {
    const { container } = render(<WeatherData />)
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with data', () => {
    const weatherData = {
      main: {
        temp: 10,
        humidity: 10,
        pressure: 10,
      },
      wind: {
        speed: 10,
      },
      weather: [
        {
          main: 'main',
          description: 'description',
          icon: 'icon',
        },
      ],
    }
    const { container } = render(<WeatherData weatherData={weatherData} />)
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with loading', () => {
    const { container } = render(<WeatherData isLoading />)
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with empty data', () => {
    const { container } = render(<WeatherData weatherData={{}} />)
    expect(container).toMatchSnapshot()
  })
})
