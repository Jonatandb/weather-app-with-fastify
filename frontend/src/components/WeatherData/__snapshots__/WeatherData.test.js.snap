// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherData should render correctly 1`] = `
<div>
  <div
    class="WeatherData-Container"
  >
    <div />
  </div>
</div>
`;

exports[`WeatherData should render correctly with data 1`] = `
<div>
  <div
    class="WeatherData-Container"
  >
    <h4
      class="WeatherData-Title"
    >
      Datos del clima üå°Ô∏è
    </h4>
    <div
      class="WeatherData-Details-Container"
    >
      <div
        class="WeatherData-Image-Container"
      >
        <img
          alt="main description 10"
          src="https://openweathermap.org/img/wn/icod@2x.png"
        />
      </div>
      <div
        class="WeatherData-Rows-Container"
      >
        <div
          class="WeatherData-Row"
        >
          <p>
            Temperatura
            :
          </p>
          <p>
            10 ¬∫C
          </p>
        </div>
        <div
          class="WeatherData-Row"
        >
          <p>
            Humedad
            :
          </p>
          <p>
            10 %
          </p>
        </div>
        <div
          class="WeatherData-Row"
        >
          <p>
            Presi√≥n
            :
          </p>
          <p>
            10 hPa
          </p>
        </div>
        <div
          class="WeatherData-Row"
        >
          <p>
            Viento
            :
          </p>
          <p>
            10 m/s
          </p>
        </div>
        <div
          class="WeatherData-Row"
        >
          <p>
            Estado
            :
          </p>
          <p>
            main 
          </p>
        </div>
        <div
          class="WeatherData-Row"
        >
          <p>
            Cielo
            :
          </p>
          <p>
            description 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherData should render correctly with empty data 1`] = `
<div>
  <div
    class="WeatherData-Container"
  >
    <div />
  </div>
</div>
`;

exports[`WeatherData should render correctly with loading 1`] = `
<div>
  <div
    class="WeatherData-Container"
  >
    <div
      class="WeatherData-LoadingMessage"
    >
      <div
        class="SpinningIcon-fast"
        speed="fast"
      >
        ‚è≥
      </div>
       
      <p>
        Cargando datos del clima...
      </p>
    </div>
  </div>
</div>
`;
